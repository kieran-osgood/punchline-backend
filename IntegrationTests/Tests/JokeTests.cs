using System.Threading.Tasks;
using GraphQL;
using GraphQL.Data;
using GraphQL.Entities.Joke;
using HotChocolate;
using HotChocolate.Execution;
using IntegrationTests.Helpers;
using Snapshooter.Xunit;
using Xunit;

namespace IntegrationTests.Tests
{
    public class JokeTests : IClassFixture<CustomWebApplicationFactory<Startup>>
    {
        [Theory]
        [InlineData(JokeLength.Small)]
        [InlineData(JokeLength.Medium)]
        [InlineData(JokeLength.Large)]
        public async Task Jokes_Length_Filters_Results(JokeLength length)
        {
            var factory = new CustomWebApplicationFactory<Startup>().WithAuthentication(TestClaimsProvider.WithUserClaims());; 
            var query = 
                $@"query Jokes {{
                  jokes(jokeLength: {length.ToString().ToUpper()}) {{
                    nodes {{
                      id
                      body
                    }}
                  }}
                }}";

            var executor = await factory.Services.GetRequestExecutorAsync();
            var request = QueryRequestBuilder
                .New()
                .AddAuthorizedUser()
                .SetQuery(query)
                .Create();
            
            var result = await executor.ExecuteAsync(request);

            Assert.Null(result.Errors);
            
            (await result.ToJsonAsync()).MatchSnapshot();
        }
    }
}