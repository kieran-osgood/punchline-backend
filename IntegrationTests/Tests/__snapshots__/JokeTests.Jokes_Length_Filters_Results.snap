{
  "data": {
    "jokes": {
      "nodes": [
        {
          "id": "Sm9rZQppMQ==",
          "body": "TEST RECORD: You\u0027re standing on my scarf."
        },
        {
          "id": "Sm9rZQppMg==",
          "body": "TEST RECORD: Would you like a jelly baby?"
        },
        {
          "id": "Sm9rZQppMw==",
          "body": "TEST RECORD: To the rational mind, nothing is inexplicable; only unexplained."
        }
      ]
    }
  }
}
